src/elements/__tests__/transform.spec.ts

```diff
@@ -0,0 +1,58 @@
+import {
+  transfromElementIn,
+  transfromElementOut,
+} from "../transformer/Transform";
+
+describe("transform", () => {
+  describe("transfromIn", () => {
+    it("should not allow elements which are the wrong type", () => {
+      // @ts-expect-error -- we should not be able to transfrom a malformed element
+      transfromElementIn("code", {});
+
+      transfromElementIn("code", {
+        assets: [],
+        // @ts-expect-error -- we should not be able to transfrom a malformed element
+        fields: { nonExistantField: "123" },
+      });
+    });
+    it("should partially transform elements with no fields", () => {
+      const codeElement = { assets: [], fields: {} };
+      const result = transfromElementIn("code", codeElement);
+
+      expect(result).toEqual({ assets: [] });
```

Comment:
Should we strip `assets` on the way in here by default, to ensure pme doesn't see them? E.g. it's fair to say the pme code element doesn't care about `assets`.

Reply:
Is the suggestion here that if we do care about them, then the element will has a custom transform?

Reply:
Yeah, I think so – difficult to imagine a `default` that knows where to put them? I may be wrong here!

Reply:
Seems fair, I'll do that.